Models/RecognitionViewModel.swift

```diff
@@ -0,0 +1,110 @@
+import CoreML
+import Vision
+import UIKit
+
+class RecognitionViewModel: ObservableObject {
+    @Published var selectedImage: UIImage?
+    @Published var recognitionResult: String = ""
+    @Published var history: [RecognitionRecord] = []
+    @Published var showingCamera = false
+    @Published var showingPhotoPicker = false
+    @Published var errorMessage: String?
+    
+    private let maxHistoryCount = 10
+    private let historyFileURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0].appendingPathComponent("history.json")
+    
+    init() {
+        loadHistory()
+    }
+    
+    func recognizeObject(in image: UIImage) {
+        guard let model = try? VNCoreMLModel(for: Resnet50(configuration: .init()).model) else {
+            errorMessage = "Nie zaladowano coreML"
+            return
+        }
+        
+        let request = VNCoreMLRequest(model: model) { [weak self] request, error in
+            guard let results = request.results as? [VNClassificationObservation],
+                  let topResult = results.first else {
+                self?.errorMessage = "Nie udalo sie rozpoznac"
+                return
+            }
+            
+            let resultText = "Objekt: \(topResult.identifier)\nPewność: \(Int(topResult.confidence * 100))%"
+            self?.recognitionResult = resultText
+            
+            let record = RecognitionRecord(
+                id: UUID(),
+                image: image,
+                result: resultText,
+                date: Date()
+            )
+            
+            DispatchQueue.main.async {
+                self?.history.insert(record, at: 0)
+                if self?.history.count ?? 0 > self?.maxHistoryCount ?? 10 {
+                    self?.history.removeLast()
+                }
+                self?.saveHistory()
+            }
+        }
+        
+        request.imageCropAndScaleOption = .scaleFit
+        
+        guard let cgImage = image.cgImage else {
+            errorMessage = "Zly obraz"
+            return
+        }
+        
+        let handler = VNImageRequestHandler(cgImage: cgImage, options: [:])
+        try? handler.perform([request])
```

Comment:
Zamiast ignorować błąd, warto go obsłużyć i pokazać w errorMessage
```suggestion
do {
    try handler.perform([request])
} catch {
    self.errorMessage = "Błąd podczas przetwarzania obrazu: \(error.localizedDescription)"
}
```